### PR TITLE
17 pixel

### PR DESCRIPTION
**Pull request by WenliZhang1102**
Thanks for the silver lining Ryan, I find the original version.

Here, I implemented rgb565 conversion and wrote the test case for it.

The test case has passed locally.

### CODE REVIEW

**Comment by AeroStun** [comment]
The Windows CIs aren't happy

**Review by AeroStun — COMMENTED** [review]
Overall I don't like the idea of having exported functions doing conversions; I much prefer having like rgb444 is implemented, completely internally

File: src/SMCE/BoardView.cpp [hunk 1]
```diff
@@ -371,6 +371,72 @@ bool FrameBuffer::read_rgb444(std::span<std::byte> buf) {
...
+/*
+ * 76543210 | 76543210
+ * GGGBBBBB   RRRRRGGG
+ */
+void rgb565ToRgb888(std::span<const std::byte> buf, std::byte * res) {
+    // read two bytes
+    // read from rgb565 then write to the rgb888
+    for (auto i = buf.begin(); i != buf.end(); ++i) {
+        std::byte low = *i;
+        *i++;
```
Comment (AeroStun) [inline src/SMCE/BoardView.cpp]: Unused unary operand result

File: src/SMCE/BoardView.cpp [hunk 2]
```diff
@@ -371,6 +371,72 @@ bool FrameBuffer::read_rgb444(std::span<std::byte> buf) {
...
+    [[maybe_unused]] std::lock_guard lk{frame_buf.data_mut};
+
+    auto* res = frame_buf.data.data();
+    rgb565ToRgb888(buf, res);
+
+    return true;
+}
+
+void rgb888ToRgb565(const std::byte* buf, std::span<std::byte> res) {
+    // read two bytes(16bits)
```
Comment (AeroStun) [inline src/SMCE/BoardView.cpp]: Comment wrong

File: test/BoardView.cpp [hunk 3]
```diff
@@ -246,3 +248,80 @@ TEST_CASE("BoardView RGB444 cvt", "[BoardView]") {
...
+ * GGGBBBBB   RRRRRGGG
+ */
+TEST_CASE("BoardView RGB565 Read Test", "[BoardView]") {
+    // convert three bytes(24bits) of rgb888 to two bytes(18bits) of rgb565
+    std::byte res[2];
+    std::byte red_bits[] = {0xFF_b, 0_b, 0_b};
+    std::byte green_bits[] = {0_b, 0xFF_b, 0_b};
+    std::byte blue_bits[] = {0x0_b, 0_b, 0xFF_b};
+    std::byte black_bits[] = {0_b, 0_b, 0_b};
+    std::byte white_bits[] = {0xFF_b, 0xFF_b, 0xFF_b};
```
Comment (AeroStun) [inline test/BoardView.cpp:258]: Use more varied values, else it we might as well only check that plain colors work

**Comment by WenliZhang1102** [comment]
> The Windows CIs aren't happy

I am using mac, dear Ryan

File: src/SMCE/BoardView.cpp (hunk 2 above)
Comment (WenliZhang1102) [reply src/SMCE/BoardView.cpp]: I am gonna check out it! The wrong comments ;) 😊

**Comment by AeroStun** [comment]
> > The Windows CIs aren't happy
> 
> I am using mac, dear Ryan

You can use CI build logs to debug the issue

File: test/BoardView.cpp (hunk 3 above)
Comment (WenliZhang1102) [reply test/BoardView.cpp:258]: I am gonna add an elegant purple or the shining green!

**Pull request merged**